tmp/generated.cpp

```diff
@@ -0,0 +1,222 @@
+struct cutlass3x_sm90_tensorop_h64x128x16gemm_f16_f16_f16_void_f16_128x128x64_2x1x1_0_ttn_align8_warpspecialized_cooperative_epi_tma :
+  public cutlass3x_sm90_tensorop_h64x128x16gemm_f16_f16_f16_void_f16_128x128x64_2x1x1_0_ttn_align8_warpspecialized_cooperative_epi_tma_base { };
+
+
+  using cutlass3x_sm90_tensorop_h64x128x16gemm_f16_f16_f16_void_f16_128x128x64_2x1x1_0_ttn_align8_warpspecialized_cooperative_epi_tma_device_type = cutlass::gemm::device::GemmUniversalAdapter<cutlass3x_sm90_tensorop_h64x128x16gemm_f16_f16_f16_void_f16_128x128x64_2x1x1_0_ttn_align8_warpspecialized_cooperative_epi_tma>;
+
+// When workspace_size is not a nullptr, populates requested workspace_size and returns.
+// Otherwise, compuates the Gemm kernel using the given workspace ptr.
+extern "C" {
+PT_EXPORT int KERNEL_NAME(const half* X, const half* W, half* Y, size_t* workspace_size, uint8_t* workspace, cudaStream_t stream) {
+  try {
+  
+  {
+    if (!X) {
+      int64_t X_size = 8192L;
+      if (X_size > 0) {
+        throw std::runtime_error("input X is null but size is not 0!");
+      }
+    }
+  }
+
+  
+  {
+    if (!W) {
+      int64_t W_size = 8192L;
+      if (W_size > 0) {
+        throw std::runtime_error("input W is null but size is not 0!");
+      }
+    }
+  }
+
+  
+  
+  {
+    if (!Y) {
+      int64_t Y_size = 65536L;
+      if (Y_size > 0) {
+        throw std::runtime_error("input Y is null but size is not 0!");
+      }
+    }
+  }
+
+  int64_t B = 1;
+  int64_t M = 256L;
+  int64_t K = 32L;
+  int64_t N = 256L;
+  using ElementComputeEpilogue = cutlass3x_sm90_tensorop_h64x128x16gemm_f16_f16_f16_void_f16_128x128x64_2x1x1_0_ttn_align8_warpspecialized_cooperative_epi_tma_device_type::ElementAccumulator;
+  using coord_t = cutlass::gemm::GemmCoord::Index;
+  cutlass3x_sm90_tensorop_h64x128x16gemm_f16_f16_f16_void_f16_128x128x64_2x1x1_0_ttn_align8_warpspecialized_cooperative_epi_tma_device_type::Arguments arguments;
+  
+  // Initialize GemmUniversal3xInstance arguments.
+  arguments = {
+    cutlass::gemm::GemmUniversalMode::kGemm,  // GemmUniversalMode mode
+    {
+      static_cast<coord_t>(M),
+      static_cast<coord_t>(N),
+      static_cast<coord_t>(K),
+      static_cast<coord_t>(B)
+    }, // ProblemShape problem_shape
+    {
+      (cutlass::half_t*)(X),  // ElementA const* ptr_A
+      {
+        32L /* stride_x0 */,
+        cute::Int<1>{} /* stride_x1 */,
+        0 /* batch_stride_x */
+      },  // StrideA dA
+      (cutlass::half_t*)(W),  // ElementB const* ptr_B
+      {
+        cute::Int<1>{} /* stride_w1 */,
+        256L /* stride_w0 */,
+        0 /* batch_stride_w */
+      },  // StrideB dB
+    },  // MainloopArguments mainloop
+    
+    // see https://github.com/NVIDIA/cutlass/blob/e0aaa3c3b38db9a89c31f04fef91e92123ad5e2e/include/cutlass/epilogue/collective/sm90_epilogue_tma_warpspecialized.hpp#L184
+    {
+      {{}, { static_cast<ElementAcc>(3.0) }},  // thread, typename FusionCallbacks::Arguments ( EVT Arguments )
```

Comment:
These are the EVT arguments (e.g the field arguments.epilogue.thread ). See https://github.com/NVIDIA/cutlass/blob/e0aaa3c3b38db9a89c31f04fef91e92123ad5e2e/examples/49_hopper_gemm_with_collective_builder/49_collective_builder.cu#L437-L464 for the a doc comment how they are built up. Their structure mirrors the structure of the EVT functor above and provides neccessary arguments.